### PR TITLE
Feature/change password functionality

### PR DESCRIPTION
**Pull request by HarisShahidHameed**
Trello link : https://trello.com/c/0YdEhms0/20-changing-password-functionality

### CODE REVIEW

File: server/libs/userlibs.js [hunk 1]
```diff
@@ -148,4 +148,35 @@ const logOut = function(request) {
...
 
-module.exports = {register, logIn, updateUser, logOut}
\ No newline at end of file
+const changePassword = function(request) {
+    let body = request.body;
+    return new Promise((resolve,reject) => {
+        bcrypt.genSalt(10, (err, salt) => {
+            bcrypt.hash(body.password, salt, (err,hash) => {
+                body.password = hash;
+                User.findOneAndUpdate({uuid : body.uuid, deleted : false}, {$set : {password : body.password}}, {new: true}).then((user) => {
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: .then should b in next line

File: server/routes/user.js [hunk 2]
```diff
@@ -35,4 +35,12 @@ router.post('/user/logout', authenticate, (req,res) => {
     })
 })
 
+router.post('/user/changePassword', authenticate, (req,res) => {
+    userLibs.changePassword(req).then((success) => {
```
Comment (abdulhaqshah) [inline server/routes/user.js]: req should not be sent as a parameter

File: server/libs/userlibs.js (hunk 1 above)
Comment (abdulhaqshah) [reply server/libs/userlibs.js]: deactivate check instead of delete

File: server/libs/userlibs.js [hunk 3]
```diff
@@ -148,4 +148,35 @@ const logOut = function(request) {
...
+                    if (user) {
+                        resolve({
+                            status : statusCodes.successful,
+                            message : `Password ${messages.updated}`, 
+                            data : user
```
Comment (abdulhaqshah) [inline server/libs/userlibs.js]: why are we sending user

Comment (abdulhaqshah) [reply server/libs/userlibs.js]: don't need to send whole user object just a new password only

**Review by abdulhaqshah — CHANGES_REQUESTED** [review]

File: server/libs/userlibs.js (hunk 3 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done

File: server/routes/user.js (hunk 2 above)
Comment (HarisShahidHameed) [reply server/routes/user.js]: Done

File: server/libs/userlibs.js (hunk 1 above)
Comment (HarisShahidHameed) [reply server/libs/userlibs.js]: Done

**Review by abdulhaqshah — APPROVED** [review]

**Pull request merged**